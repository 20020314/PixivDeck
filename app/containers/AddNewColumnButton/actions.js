// @flow
import type { User } from '../../types/user'
import type { Action } from './actionTypes.js'
import { ADD_USER_ILLUST } from './constants'

export function addUserIllust(user: User): Action {
  return {
    type: ADD_USER_ILLUST,
    user,
  }
}
