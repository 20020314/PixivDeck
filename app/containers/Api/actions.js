// @flow
import type { Response } from 'services/api'

import { API_REQUEST_SUCCESS } from './actionTypes'
import type { ApiRequestSuccess } from './actionTypes'

export function apiRequestSuccess(response: Response): ApiRequestSuccess {
  return {
    type: API_REQUEST_SUCCESS,
    response,
  }
}
