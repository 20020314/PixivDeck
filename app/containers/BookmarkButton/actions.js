// @flow
import type { Restrict } from './types'

import {
  ADD_BOOKMARK_REQUEST,
  ADD_BOOKMARK_SUCCESS,
  ADD_BOOKMARK_FAILURE,
  DELETE_BOOKMARK_REQUEST,
  DELETE_BOOKMARK_SUCCESS,
  DELETE_BOOKMARK_FAILURE,
} from './actionTypes'
import type {
  AddBookmarkRequest,
  AddBookmarkSuccess,
  AddBookmarkFailure,
  DeleteBookmarkRequest,
  DeleteBookmarkSuccess,
  DeleteBookmarkFailure,
} from './actionTypes'

export function addBookmarkRequest(
  id: number,
  restrict: Restrict
): AddBookmarkRequest {
  return {
    type: ADD_BOOKMARK_REQUEST,
    id,
    restrict,
  }
}
export function addBookmarkSuccess(
  id: number,
  restrict: Restrict
): AddBookmarkSuccess {
  return {
    type: ADD_BOOKMARK_SUCCESS,
    id,
    restrict,
  }
}
export function addBookmarkFailure(
  id: number,
  error: string
): AddBookmarkFailure {
  return {
    type: ADD_BOOKMARK_FAILURE,
    id,
    error,
  }
}
export function deleteBookmarkRequest(id: number): DeleteBookmarkRequest {
  return {
    type: DELETE_BOOKMARK_REQUEST,
    id,
  }
}
export function deleteBookmarkSuccess(id: number): DeleteBookmarkSuccess {
  return {
    type: DELETE_BOOKMARK_SUCCESS,
    id,
  }
}
export function deleteBookmarkFailure(
  id: number,
  error: string
): DeleteBookmarkFailure {
  return {
    type: DELETE_BOOKMARK_FAILURE,
    id,
    error,
  }
}
