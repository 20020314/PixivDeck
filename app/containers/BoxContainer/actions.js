// @flow
import type { Action } from './actionTypes.js'
import { OPEN_PIXIV } from './constants'

export function openPixiv(id: number): Action {
  return {
    type: OPEN_PIXIV,
    id,
  }
}
