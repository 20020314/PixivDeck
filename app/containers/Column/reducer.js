// @flow
import update from 'utils/update'
import { REHYDRATE } from 'redux-persist/constants'
import { handleRehydrate } from 'utils/handleReydrate'

export type BaseColumn = {
  ids: Array<number>,
  nextUrl: ?string,
}

type ActionTypes = {
  ADD_COLUMN_SUCCESS: $Subtype<string>,
  SET_NEXT_URL: $Subtype<string>,
  FETCH_SUCCESS: $Subtype<string>,
  FETCH_NEXT_SUCCESS?: $Subtype<string>,
  FETCH_NEW_SUCCESS?: $Subtype<string>,
}

export function baseReducer<
  T: BaseColumn,
  S: { [*]: T },
  AS: ActionTypes,
  A: { +type: $Subtype<string> }
>(name: string, types: AS, state: S, action: A): S {
  switch (action.type) {
    case types.SET_NEXT_URL:
      return update(state, action, { nextUrl: action.nextUrl })
    case types.ADD_COLUMN_SUCCESS:
      return update(state, action, { ids: [], nextUrl: null })
    case types.FETCH_SUCCESS:
    case types.FETCH_NEXT_SUCCESS:
    case types.FETCH_NEW_SUCCESS:
      return update(state, action, { ids: action.ids })
    case REHYDRATE:
      return handleRehydrate(state, action, name)
    default:
      return state
  }
}
