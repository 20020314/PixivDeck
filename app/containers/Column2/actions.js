// @flow
import type { Action } from './actionTypes.js'
import { ADD_COLUMN } from './constants'
import type { Body } from './reducer'

export function addColumn(id: string, body: Body): Action {
  return {
    type: ADD_COLUMN,
    id,
    body,
  }
}
