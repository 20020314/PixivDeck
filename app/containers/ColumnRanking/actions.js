// @flow
import type { ColumnId } from '../ColumnManager/reducer'
import type { Response } from '../../api/schema'
import type { Action } from './actionTypes.js'
import {
  ADD_RANKING_COLUMN,
  ADD_RANKING_COLUMN_SUCCESS,
  FETCH_RANKING,
  FETCH_RANKING_FAILRE,
  SET_NEXT_URL,
  FETCH_RANKING_SUCCESS,
} from './constants'
import type { Mode } from './reducer'

export function addRankingColumn(mode: Mode): Action {
  return {
    type: ADD_RANKING_COLUMN,
    mode,
  }
}

export function addRankingColumnSuccess(
  id: ColumnId,
  mode: Mode,
  title: string
): Action {
  return {
    type: ADD_RANKING_COLUMN_SUCCESS,
    id,
    mode,
    title,
  }
}

export function fetchRanking(id: ColumnId): Action {
  return {
    type: FETCH_RANKING,
    id,
  }
}

export function fetchRankingFailre(id: ColumnId): Action {
  return {
    type: FETCH_RANKING_FAILRE,
    id,
  }
}

export function setNextUrl(id: ColumnId, nextUrl: string): Action {
  return {
    type: SET_NEXT_URL,
    id,
    nextUrl,
  }
}

export function fetchRankingSuccess(
  id: ColumnId,
  response: Response,
  ids: Array<string>
): Action {
  return {
    type: FETCH_RANKING_SUCCESS,
    id,
    response,
    ids,
  }
}
