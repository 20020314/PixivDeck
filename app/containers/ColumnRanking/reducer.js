// @flow
import type { Action } from './actionTypes'
import * as Actions from './constants'
import { REHYDRATE } from 'redux-persist/constants'

export type Mode =
  | 'day'
  | 'week'
  | 'month'
  | 'day_male'
  | 'day_female'
  | 'week_original'
  | 'week_rookie'

export type Endpoint = '/v1/illust/ranking'

export type ColumnId = Mode

export type ColumnRanking = {|
  illustIds: Array<number>,
  nextUrl: ?string,
|}

export type State = $Shape<{ [Mode]: ColumnRanking }>

const initialState: State = {}

export default function(state: State = initialState, action: Action): State {
  switch (action.type) {
    case Actions.ADD_RANKING_COLUMN_SUCCESS:
      return { ...state, [action.id]: { illustIds: [], nextUrl: null } }

    case Actions.SET_NEXT_URL: {
      const id = action.id
      return {
        ...state,
        [id]: { ...state[id], nextUrl: action.nextUrl },
      }
    }

    case Actions.FETCH_RANKING_SUCCESS:
    case Actions.FETCH_NEXT_RANKING_SUCCESS: {
      const id = action.id
      return {
        ...state,
        [id]: { ...state[id], illustIds: action.ids },
      }
    }

    // TODO REHYDRATEをまとめる
    case REHYDRATE: {
      // $FlowFixMe
      const oldState = action.payload.ColumnRanking

      if (oldState) {
        const newState = Object.keys(oldState).reduce((acc, key) => {
          acc[key] = {
            illustIds: [],
            nextUrl: null,
          }
          return acc
        }, {})
        return newState
      }
      return state
    }

    default:
      return state
  }
}
