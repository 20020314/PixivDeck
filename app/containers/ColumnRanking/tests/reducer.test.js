// @flow
import reducer, { initialState } from '../reducer'
import * as actions from '../actions'

test('default action', () => {
  // $FlowFixMe
  expect(reducer(initialState, { type: 'default action' })).toMatchSnapshot()
})

test('ADD_COLUMN_SUCCESS', () => {
  expect(
    reducer(initialState, actions.addColumnSuccess('day'))
  ).toMatchSnapshot()
})
