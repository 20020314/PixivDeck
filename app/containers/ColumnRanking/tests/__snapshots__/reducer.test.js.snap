// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ADD_COLUMN_SUCCESS 1`] = `
Object {
  "day": Object {
    "ids": Array [],
    "interval": 10800000,
    "nextUrl": null,
  },
}
`;

exports[`default action 1`] = `Object {}`;
